### PR TITLE
[Layout foundations] Remove default spacing from `Inline`

### LINKED ISSUE #8368: [Layout foundations] Remove default spacing from `Inline`
As per https://github.com/Shopify/polaris/discussions/8329
- [ ] Remove default `gap` of `4` and update to `0`
- [ ] Update usages in Polaris as required

### PR DESCRIPTION
**Pull request by chazdean**
### WHY are these changes introduced?

Fixes #8368 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Review by kyledurand — APPROVED** [review]
LGTM

File: .changeset/eight-spies-complain.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@shopify/polaris': minor
+'polaris.shopify.com': minor
+---
+
+Remove default spacing from `Inline`
```
Comment (laurkim) [inline .changeset/eight-spies-complain.md]:
```suggestion
Removed default spacing from `Inline`
```

**Review by laurkim — APPROVED** [review]
Nice 💯

**Review by aveline — APPROVED** [review]
The change looks good, but I found it interesting that no tests had to be updated. I think `Inline` could use a couple more test cases. Could we do that as part of this PR?

**Pull request merged**